### DISCUSSION TITLE
unable to get this to work right in Unraid and connect to printers

### DISCUSSION THREAD

**Discussion by comet424** [mainsail-crew/mainsail#1798] [Help]
hi so i installed  mainsail docker from CA  store...  i copied the  config.json..  into the appdata folder..  and started things... i eded the moonraker.config file on the printers i have and added the ip address of the unraid and port  under cors_domains:  from the info it stated....  i added printer..  and 7125 port  it trys to connect but it wont connect...  what am i doing room...  and the remotemode documation link doesnt work

**Comment by meteyou** [comment]
Pls Post your Json file and your moonraker.conf file. Pls don't use any moonraker.conf file, because you overwrite important/old settings with that! Pls also post a screenshot from your error message.

And pls select one channel you want to use. Don't ask the same question are and in the discord. The same people answer here and on our discord server. So it's only wasted time for our crew to ask the same question on multiple channels.

**Reply by comet424** [reply]
as for the documentation from the bot that replied.. it said if it doesnt state unraid in there then no one will reply in in discord  and please write your question in the repo...  so thats what i did.. since there is no documentation for unraid,, and the documentation said it will get ignored just post in the repo...  but ok

and i had this posted as a bug  but it got demed not a bug
config.json
```
{
  "remoteMode": true
}
```

```[server]
host: localhost
port: 7125
enable_debug_logging: False
klippy_uds_address: /tmp/klippy_uds

[authorization]
trusted_clients:
    10.0.0.0/8
    127.0.0.0/8
    169.254.0.0/16
    172.16.0.0/12
    192.168.0.0/16
    FE80::/10
    ::1/128
cors_domains:
    http://*.lan
    http://*.local
    https://my.mainsail.xyz
    http://my.mainsail.xyz
    https://app.fluidd.xyz
    http://app.fluidd.xyz
    http://192.168.0.3:8099
    
[database]
database_path: /home/mks/.moonraker_database

[file_manager]
config_path: /home/mks/klipper_config
log_path: /home/mks/klipper_logs

[octoprint_compat]

[history]

```

i did what the popup said  i added the 
    http://192.168.0.3:8099
but it didnt work

![image_2024-03-02_073001688](https://github.com/mainsail-crew/mainsail/assets/75998611/b7287bb5-ec99-432c-8d3a-a2c5313f8dee)


and the other error for the remote mode documentation is
![image_2024-03-02_075049713](https://github.com/mainsail-crew/mainsail/assets/75998611/f9edb9b7-7a44-4002-8d46-e9fd7b34a03a)

![image_2024-03-02_075209673](https://github.com/mainsail-crew/mainsail/assets/75998611/ff405e91-10a4-431f-a20d-7d233e1bd48e)

**Reply by meteyou** [reply]
pls point me to this information:

> as for the documentation from the bot that replied.. it said if it doesnt state unraid in there then no one will reply in in discord and please write your question in the repo... so thats what i did.. since there is no documentation for unraid,, and the documentation said it will get ignored just post in the repo... but ok

i checked the bot message again and it only said, that you have to provide logs... nothing more.

**Reply by meteyou** [reply]
your moonraker config looks complete wrong to me. pls upload your moonraker.log. i think it doesn't start, because it is just wrong. so mainsail cannot connect to a not running moonraker instance. (that is my current assumption)

**Reply by meteyou** [reply]
ahh and the page you are looking for is https://docs.mainsail.xyz/overview/quicktips/config.json. It looks like a broken redirect in the docs. so your config.json is also very outdated and wrong.

**Comment by comet424** [comment]
well thats the default moonraker for the Elegoo Neptune 4 Max  so  i know nothing  all i did was do what the pop up said..
as for the bot it posted the doc link
the doc said
```
Rules
Subject of the issue
Is your issue connected to some part of the Mainsail universe or from a third-party. As already explained before, we cannot know everything. The main focus of the support is for following programs:

All other questions regarding third-party programs and third-party macros should be asked at the repo of the project and might get ignored.
```

so i read it as  since unraid is not part of the listing..  then no one will help me in discord.. so i never went further in the documentation as i took it no one will help me in discord  it will get ignored and i need to write it in  repo .. thats my understanding

here is the moonraker log of one of the printers  do you need all 4 of them?
[moonrakerlog.txt](https://github.com/mainsail-crew/mainsail/files/14469904/moonrakerlog.txt)


the config.json file is from what i got from the unraid.. so whoever is responsible for the unraid repo Mainsail   i guess is person to fix that?
i got that info sinec it says

```
Mainsail makes Klipper more accessible by adding a lightweight, responsive web user interface, centred around an intuitive and consistent design philosophy.


Download config.json BEFORE you start your container
https://raw.githubusercontent.com/patrickstigler/unraid_app_templates/main/mainsail/config.json


And put it in your config folder, the default is /mnt/user/appdata/mainsail/config.json


Happy printing


Additional Requirements:
None Listed
```


so i just trying to follow the instructions   but thats all i know...   let me know what else i need to know

**Comment by comet424** [comment]
i do know if i need to print in Cura  i have to install the Moonraker plugin... but in Oracslicer  i dont have to install anything  thats all i know

**Reply by meteyou** [reply]
is this a question or only a side information?

**Reply by comet424** [reply]
side information    that all i know that works

**Comment by meteyou** [comment]
> so i read it as since unraid is not part of the listing.. then no one will help me in discord.. so i never went further in the documentation as i took it no one will help me in discord it will get ignored and i need to write it in repo .. thats my understanding

but this is the same on discord or in our repo discussions... but this mean we cannot help you to setup unraid or answer on unraid specific topics, but we can help you with mainsail, moonraker and klipper.

> here is the moonraker log of one of the printers do you need all 4 of them?
[moonrakerlog.txt](https://github.com/mainsail-crew/mainsail/files/14469904/moonrakerlog.txt)

this link doesnt work... but now the first moonraker.log would fit to find the issue...

> the config.json file is from what i got from the unraid.. so whoever is responsible for the unraid repo Mainsail i guess is person to fix that?
i got that info sinec it says

the config.json is a mainsail specific file, but your guide which one you use is just outdated. the link is about 2 years old. so the guy how uploaded this file should fix it.

but i'm only wondering, unraid can use docker containers? why do you don't use the official docker container?

**Reply by comet424** [reply]
so for the repo discussion i didnt know  that the same people are in both... way i read it and understood...  if like unraid wasnt in the docs.  then no one in the discord can help me  all i can do is post my help in the repo on the ghithub.. that was my understanding  as not everyone is on discord.. etc..   but now i know

ill upload the log 
[moonrakerlog.txt](https://github.com/mainsail-crew/mainsail/files/14470031/moonrakerlog.txt)

as for the 2 year old file.. i guess the person that wrote the mainsail program  needs to fix the config file?  since its linked  mainsale... so the support page and the project page go to main sail.. so any issues with the unraid docker  and its mainsail issue...

as for offical docker container.. well you said it.. the offical docker container is 2 years old....  i just install it  and thats it 

ill do screen shots
![unraidmain](https://github.com/mainsail-crew/mainsail/assets/75998611/52063cbe-d4cf-4a5c-87f2-dc3d27c2cd55)
![unraidmain2](https://github.com/mainsail-crew/mainsail/assets/75998611/966e4160-c652-4b1e-8f8a-6a1b8e25ec71)


so thats the official docker container... and any support goes to to https://docs.mainsail.xyz  
so if there i dunno more... and if its not offical anymore then i lost 

as i trying to link all 4 printers together

**Reply by meteyou** [reply]
looks like only the description in unraid is outdated and this plugin download the official docker container. so pls update the config.json with the link i wrote above.

**Comment by comet424** [comment]
ill try again  it wont allow me to up like 4 meg log file
[moonrakerl.log](https://github.com/mainsail-crew/mainsail/files/14470052/moonrakerl.log)

**Comment by comet424** [comment]
ugh my other post didnt work

**Comment by comet424** [comment]
so from what i knew  was i didnt know  discord people were on the repo board... so thats why i wrote on here. too.. since not everyone on here or there..

as for the config and stuff...  that is the offical mainsail  docker unraid container...   support page goes to the docs page.. there is no support page to talk to the person that uploaded i assume hes the person that wrote the mainsail program and forgot to fix unraid  template... or what not ..  its all i know

![unraidmain](https://github.com/mainsail-crew/mainsail/assets/75998611/fabba796-ff4a-4fc9-b145-a2a002863985)
![unraidmain2](https://github.com/mainsail-crew/mainsail/assets/75998611/e08c0e02-fb18-41b4-a250-cae80ca00666)

**Comment by comet424** [comment]
since the log file wont upload here is some of it 
```
--------------------Moonraker Log Start--------------------
config_file: /home/mks/klipper_config/moonraker.conf
log_file: /home/mks/klipper_logs/moonraker.log
software_version: ?
python_version: 3.7.3 (default, Jan 22 2021, 20:04:44)  [GCC 8.3.0]

#################### Moonraker Configuration ####################

[server]
host = localhost
port = 7125
enable_debug_logging = False
klippy_uds_address = /tmp/klippy_uds

[authorization]
trusted_clients = 
	10.0.0.0/8
	127.0.0.0/8
	169.254.0.0/16
	172.16.0.0/12
	192.168.0.0/16
	FE80::/10
	::1/128
cors_domains = 
	http://*.lan
	http://*.local
	https://my.mainsail.xyz
	http://my.mainsail.xyz
	https://app.fluidd.xyz
	http://app.fluidd.xyz

[database]
database_path = /home/mks/.moonraker_database

[file_manager]
config_path = /home/mks/klipper_config
log_path = /home/mks/klipper_logs

[octoprint_compat]

[history]

#################################################################
All Configuration Files:
/home/mks/klipper_config/moonraker.conf
#################################################################
Unsafe Shutdown Count: 0

System Info:

***python***
  version: sys.version_info(major=3, minor=7, micro=3, releaselevel='final', serial=0)
  version_string: 3.7.3 (default, Jan 22 2021, 20:04:44)  [GCC 8.3.0]

***cpu_info***
  cpu_count: 4
  bits: 64bit
  processor: aarch64
  cpu_desc: 
  serial_number: 
  hardware_desc: 
  model: 
  total_memory: 999784
  memory_units: kB

***sd_info***

***distribution***
  name: Armbian 22.05.0-trunk Buster
  id: debian
  version: 10
  version_parts: {'major': '10', 'minor': '', 'build_number': ''}
  like: 
  codename: buster
  release_info: {'name': '# PLEASE DO NOT EDIT THIS FILE', 'id': 'armbian'}

***virtualization***
  virt_type: none
  virt_identifier: none
Klipper Version: ?
2024-03-02 00:00:02,988 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:00:02,996 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:00:03,003 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:00:12,267 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:00:12,273 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:00:12,280 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:00:21,563 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:00:21,572 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:00:30,839 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:00:30,846 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:00:30,852 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:00:40,135 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:00:40,139 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:00:40,145 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:00:49,643 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:00:49,650 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:00:58,927 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:00:58,930 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:00:58,937 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:01:08,203 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:01:08,209 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:01:08,215 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:01:17,519 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:01:17,522 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:01:17,528 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:01:26,803 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:01:26,806 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:01:26,813 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:01:36,063 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:01:36,066 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:01:36,073 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:01:45,367 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:01:45,373 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
2024-03-02 00:01:45,379 [file_manager.py:complete_file_write()] - Invalid file write event: saved_variables.cfg
```

this is one of the printers

**Comment by comet424** [comment]
so if its outdated 2 years.. does that mean i unable to get this mainsail program to work then.. possible user is dead  or no longer online if he didnt write it and no way to get it working then?

**Comment by comet424** [comment]
try this...  i put the log file on  my One Drive 
[https://1drv.ms/f/s!AkZ74ugB21kngYgYBlUX3fAswPeUZA?e=ZbJT1R](url)

**Comment by meteyou** [comment]
pls open `http://192.168.0.172:7125` and check if you see the moonraker page (you can change the ip to every printer you have to double check all printers)

**Reply by comet424** [reply]
i get page cant be found for all 4 printers... but i i can access those printers on port 80  if thats what you mean  i also updated that link.. i uploaded 4 printer klipper logs  4 moonraker logs and 4 moonraker confs   so you goto my one drive youll be able to read the files


and i dont know if this is a moonraker page.. i just know it as a printer page 
![image_2024-03-02_122418541](https://github.com/mainsail-crew/mainsail/assets/75998611/5243d306-2d71-4554-add3-3b0b4dcaae7c)

**Comment by comet424** [comment]
so at the beginning screen for adding the printers  when adding ip and port if i dont use 7125  but use port 80 instead... it connects that way... i take it then that above screen is the moonraker?  im still new at 3d printing    and i just click and print but always learning


but does it mean something still broken?  why is it in the moonraker.conf  saying port 7125 but that doesnt work but port 80 works

**Comment by meteyou** [comment]
no. this above is not moonraker. but i think i found your issue. in your moonraker.conf is this setting:

```yaml
[server]
host: localhost
```

so this mean, that moonraker is only local available, but the default setting in mainsail and fluidd (the screenshot above) is to redirect/proxy moonraker to port 80 in nginx.

so pls change the line `host: localhost` to `host: 0.0.0.0` and restart moonraker on your printer. then you should be able to open moonraker with `http://<ip>:7125`. after that you should be also able to connect with your mainsail instance on your unread server to connect to this printer.

**Comment by comet424** [comment]
that seemed to work... what does the 0.0.0.0  mean?

**Reply by meteyou** [reply]
at first 0.0.0.0 is the default setting of Moonraker (https://moonraker.readthedocs.io/en/latest/configuration/#server) and it mean "listen to all interfaces and all ip addresses". the old value before (localhost) means, moonraker should only listen to localhost requests.

**Comment by comet424** [comment]
ah ok... i was trying the other setting... to change the config.json   to pre populate .... but it doesnt work
config.json
```
{
  "instancesDB": "json",
  "instances": [
    { "Printer #1": "192.168.0.189", "port": 7125 },
    { "Printer #2": "192.168.0.190", "port": 7125 },
    { "Printer #3": "192.168.0.191", "port": 7125 },
    { "Printer #4": "192.168.0.172", "port": 7125 }
  ]
}
```

and the error i get is
![image_2024-03-03_091521868](https://github.com/mainsail-crew/mainsail/assets/75998611/e6ee2dfb-f39f-4956-adc1-3ab70bf7c4fd)

**Reply by meteyou** [reply]
your instances list is just wrong.... pls follow the guide exectly. you have to use "host" and "port" and don't change the name of these attributes...
